### PR TITLE
MGDCTRS-1183: compute traits checksum

### PR DESCRIPTION
**Pull request by lburgazzoli**

### CODE REVIEW

**Review by valdar — APPROVED** [review]

File: cos-fleetshard-it/cucumber/src/main/java/org/bf2/cos/fleetshard/it/cucumber/ConnectorContext.java [hunk 1]
```diff
@@ -124,12 +124,7 @@ private Map<String, Object> getPlaceholders() {
             }
             if (null != connector().getMetadata()) {
                 placeholders.put(COS_MANAGED_CONNECTOR_NAME, connector().getMetadata().getName());
-
-                if (connector().getMetadata().getLabels().containsKey(LABEL_UOW)) {
-                    placeholders.put(
-                        COS_DEPLOYMENT_UOW,
-                        connector().getMetadata().getLabels().get(LABEL_UOW));
-                }
+                placeholders.put(COS_TRAITS_CHECKSUM, Resources.computeTraitsChecksum(connector()));
```
Comment (valdar) [inline cos-fleetshard-it/cucumber/src/main/java/org/bf2/cos/fleetshard/it/cucumber/ConnectorContext.java:127]: Here you are calculating the checksum of the `managedConnector` that is 0 and then you compare that with the one calculated for the `kameletBinding`

File: cos-fleetshard-operator-camel-it/src/test/resources/CamelConnectorReify.feature [hunk 2]
```diff
@@ -59,7 +59,7 @@ Feature: Camel Connector Reify
           | app.kubernetes.io/instance            | ${cos.deployment.id}          |
      And the klb has an array at path "$.spec.integration.configuration" containing:
           | { "type":"secret" , "value": "${json-unit.ignore}" }                         |
-          | { "type":"env" , "value": "CONNECTOR_DEPLOYMENT_UOW=${cos.deployment.uow}" } |
+          | { "type":"env" , "value": "CONNECTOR_TRAITS_CHECKSUM=${cos.traits.checksum}" } |
```
Comment (valdar) [inline cos-fleetshard-operator-camel-it/src/test/resources/CamelConnectorReify.feature:62]: See here...

**Review by valdar — CHANGES_REQUESTED** [review]

File: cos-fleetshard-it/cucumber/src/main/java/org/bf2/cos/fleetshard/it/cucumber/ConnectorContext.java (hunk 1 above)
Comment (lburgazzoli) [reply cos-fleetshard-it/cucumber/src/main/java/org/bf2/cos/fleetshard/it/cucumber/ConnectorContext.java:127]: yeah, that's wrong

Comment (lburgazzoli) [reply cos-fleetshard-it/cucumber/src/main/java/org/bf2/cos/fleetshard/it/cucumber/ConnectorContext.java]: fixed

File: cos-fleetshard-operator-camel-it/src/test/resources/CamelConnectorReify.feature (hunk 2 above)
Comment (lburgazzoli) [reply cos-fleetshard-operator-camel-it/src/test/resources/CamelConnectorReify.feature:62]: fixed

**Review by valdar — APPROVED** [review]

**Pull request merged**